reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,214 @@
+
+You can filter maps using `.?`. For example, to filter a list of stages by type,
+you would use `${execution["stages"].?[type == "bake"]}`.
+
+### Math
+
+You can use arithmetic operations in your expressions, such as
+`${trigger["buildInfo"]["number"] * 2}`.
+
+### Strings
+
+Strings evaluate to themselves: `${"This is a String."}` becomes "This is a
+String."
+
+## Helper properties
+
+[Helper properties](/guides/user/pipeline/expressions/#helper-properties) are
+attribute shortcuts that you can use in Spinnaker. The specific properties are:
+
+* `execution`: refers to the current pipeline execution.
+* `parameters`: refers to pipeline parameters.
+* `trigger`: refers to the pipeline trigger.
+* `scmInfo`: refers to the git details of either the trigger or the most
+recently executed Jenkins stage.
+  * `scmInfo.sha1` returns the hash of the last build
+  * `scmInfo.branch` returns the branch name
+* `deployedServerGroups`: refers to the server group that was created by the
+  last deploy stage. It should look something like:
+  `{"account":"my-gce-account",
+    "capacity": { "desired":1.0,"max":1.0,"min":1.0},
+    "region":"us-central1",
+    "serverGroup":"gcespin-dev-v005"}`
+
+## Helper functions
+
+### #alphanumerical(String)
+
+Returns the alphanumerical value of the passed-in string. That is, the input
+string with all characters aside from A-Z and 0-9 stripped out.
+
+### #readJson(String)
+
+Converts a JSON String into a Map that can then be processed further.
+
+### #fromUrl(String)
+
+Returns the contents of the specified URL as a String.
+
+### #jsonFromUrl(String)
+
+Retrieves the contents of the given URL and converts it into either a map or a
+list. You can use this to fetch information from the Spinnaker API or other
+sources.
+
+### #judgment(String)
+
+Returns the selected judgment value from the Manual Judgment stage whose name
+matches the input string. Note that `#judgment` is case sensitive:
+`${#judgment("my manual judgment stage")}` returns an error if your stage is
+named _"My Manual Judgment Stage"_.
+
+### #propertiesFromUrl(String)
+
+Retrieves the contents of the properties file at the given URL and converts it
+into a map. You can use this to fetch information from Jenkins properties files
+or other similar endpoints.
+
+### #stage(String)
+
+A shortcut to get the stage by name. For example, `${#stage("Bake")}` allows
+you to access your Bake stage. Note that `#stage` is case sensitive: if your
+stage is actually named "_bake_", `${#stage("Bake")}` will not find it. Remember
+that the values for the stage are still under the context, so you would access a
+property via `${#stage("Bake")["context"]["desiredProperty"]}`.
+
+### #toBoolean(String)
+
+Converts the input string to a boolean.
+
+### #toFloat(String)
+
+Converts a value to a floating point number.
+
+### #toInt(String)
+
+Converts a value to an integer.
+
+### #toJson(Object)
+
+Converts a map into a JSON string.
```

Comment:
This converts an arbitrary object so not necessarily a map. It can be the results of a jsonFromUrl for example. It is useful to pass more complex data structures into downstream systems, such as a Jenkins or Run Job stage parameter

Reply:
An arbitrary JSON object, right?